### DISCUSSION TITLE
Incorrect values in `DstAS` field in Akvorado

### DISCUSSION THREAD

**Discussion by hohl** [akvorado/akvorado#514] [Q&A]
I noticed some odd ASNs show up as `DstAS` in Akvorado a few weeks ago. Tiny ASNs I never heard of popped up but with lots of traffic according to Akvorado. The reported traffic made absolutely no sense, bit it didn't take much digging to figure out what was going on: displaying the related `DstNetPrefix` for those ASNs, showed completely unrelated prefixes. Unlike the ASNs, the prefixes made much more sense. 

Displaying the `DstASPath` next to the `DstAS` made the wrong data in the `DstAS` field even more obvious. For example:

![Miss-labeled sample 1](https://user-images.githubusercontent.com/744417/220554606-cb7a6ea8-31ba-4c39-b7fd-a0c805f26bbe.png)

How does Akvordo populate the `DstAS` field? Apparently not through the last element of `DstASPath`. 

I also couldn't really spot any pattern in the miss-labeled ASN yet. It happens to peers as well as transit networks, it happens to far away as well as close networks, it happens to prefixes received from different routers, etc.

Any ideas where I could start debugging? Sounds to me like some _IP-to-ASN_ database got messed up, but that's just using the Docker version of Akvorado, there isn't even as much as I could have messed up myself. Can I flush/force reload that database somehow?

**Comment by vincentbernat** [comment]
It depends. Are you using sFlow or BMP to get the AS path? If you are using sFlow, the destination AS should be the last element of the path if there is a path, unless you did configure `asn-providers` differently.

**Reply by hohl** [reply]
No sFlow involved but IPFIX and BMP, both fed in from multiple Juniper edge routers. 

`asn-providers` hasn't been set in the YAML file and according to the docs this should imply a default of `[ flow, bmp, geoip ]`.

However, I gave it a try and reversed those values to `[ geoip, bmp, flow ]`. Seems it no longer happens. So, I assume the wrong data for `DstAS` comes from IPFIX?

**Reply by vincentbernat** [reply]
It may be possible. You could check if you export destination AS in IPFIX with a capture. Otherwise, it may mean there is a bug in the BMP implementation.
